bioptim/dynamics/problem.py

```diff
@@ -391,7 +437,7 @@ def configure_qdot(nlp, as_states: bool, as_controls: bool):
         if as_states:
             nlp.x = vertcat(nlp.x, qdot)
             nlp.var_states["qdot"] = nlp.shape["qdot"]
-            qdot_bounds = nlp.x_bounds[nlp.shape["q"] :]
+            qdot_bounds = nlp.x_bounds[nlp.shape["q"]:]
```

Comment:
Please note that the shape of q is not necessarily equal to shape of qdot. Use the proper ones